src/components/LottoGame/index.js

```diff
@@ -1,107 +1,90 @@
+  const publishLottoTickets = () => {
+    const amountOfLottoTicket = purchaseAmount / UNIT_AMOUNT;
     const lottoTickets = Array(amountOfLottoTicket)
       .fill()
       .map(() => generateLottoNumbers());
 
-    this.setState({ lottoTickets });
-  }
-
-  setResultNumbers(resultNumbers) {
-    this.setState({ resultNumbers });
-  }
+    setLottoTickets(lottoTickets);
+  };
 
-  restartGame() {
-    this.setState({
-      ...this.initState,
-    });
-  }
+  const buyLottoTickets = purchaseAmount => {
+    setPurchaseAmountSubmitted(true);
+    setPurchaseAmount(purchaseAmount);
+    publishLottoTickets();
+  };
 
-  openResultModal() {
-    this.setState({ isModalOpened: true });
-  }
+  const restartGame = () => {
+    setPurchaseAmount('');
+    setPurchaseAmountSubmitted(false);
+    setLottoTickets([]);
+    setResultNumbers({ winningNumbers: [], bonusNumber: 0 });
+    setModalOpened(false);
+  };
 
-  closeResultModal() {
-    this.setState({ isModalOpened: false });
-  }
+  const openResultModal = () => {
+    setModalOpened(true);
+  };
 
-  render() {
-    const {
-      purchaseAmount,
-      lottoTickets,
-      resultNumbers: { winningNumbers, bonusNumber },
-    } = this.state;
-    const lottoResult = getLottoResult(purchaseAmount, lottoTickets, winningNumbers, bonusNumber);
+  const closeResultModal = () => {
+    setModalOpened(false);
+  };
 
-    return (
-      <>
-        <div className="flex justify-center mt-5">
-          <div className="w-full">
-            <h1 className="text-center">🎱 행운의 로또</h1>
-            <PurchaseAmountForm
-              purchaseAmount={this.state.purchaseAmount}
-              isPurchaseAmountSubmitted={this.state.isPurchaseAmountSubmitted}
-              handleChange={this.handleChange}
-              publishLottoTickets={this.publishLottoTickets}
-            />
-            {this.state.isPurchaseAmountSubmitted && <AnnouncementTime />}
-            {this.state.isPurchaseAmountSubmitted && <LottoTicketList lottoTickets={this.state.lottoTickets} />}
-            {this.state.isPurchaseAmountSubmitted && (
-              <LottoResultForm setResultNumbers={this.setResultNumbers} openResultModal={this.openResultModal} />
-            )}
-          </div>
+  return (
+    <>
+      <div className="flex justify-center mt-5">
+        <div className="w-full">
+          <h1 className="text-center">🎱 행운의 로또</h1>
+          <PurchaseAmountForm
+            purchaseAmount={purchaseAmount}
+            isPurchaseAmountSubmitted={isPurchaseAmountSubmitted}
+            handleChange={handleChange}
+            buyLottoTickets={buyLottoTickets}
+          />
+          {isPurchaseAmountSubmitted && <AnnounceTime />}
+          {isPurchaseAmountSubmitted && <LottoTicketList lottoTickets={lottoTickets} />}
+          {isPurchaseAmountSubmitted && (
+            <LottoResultForm setResultNumbers={setResultNumbers} openResultModal={openResultModal} />
+          )}
```

Comment:
여러번 분기를 해줘야 할 이유가 있나요?

```suggestion
          {isPurchaseAmountSubmitted ?
          <>
            <AnnounceTime />}
            <LottoTicketList lottoTickets={lottoTickets} />}
            <LottoResultForm setResultNumbers={setResultNumbers} openResultModal={openResultModal} />
          </> : null
          )}
```

Reply:
step1에 질문드렸던 내용인데 반영을 안 했네요 😂
제안해주신 예시대로 코드를 변경하였습니다!